File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        sum = 0
        product = 1
        newVar = n

        while newVar != 0:
            temp = newVar % 10
            sum += temp
            product *= temp
            newVar //= 10

        newSum = sum + product

        return n % newSum == 0

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int sum = 0;
        int product = 1;
        int newVar = n;

        while (newVar != 0) {
            int temp = newVar % 10;
            sum += temp;
            product *= temp;
            newVar /= 10;
        }

        int newSum = sum + product;

        return n % newSum == 0;
    }
};